pkgs/by-name/im/imhex/package.nix

```
@@ -60,29 +62,33 @@ stdenv.mkDerivation rec {
     mbedtls
     nlohmann_json
     yara
+  ] ++ lib.optionals stdenv.isDarwin [
+    darwin.apple_sdk_11_0.frameworks.UniformTypeIdentifiers
   ];
 
   cmakeFlags = [
-    "-DIMHEX_OFFLINE_BUILD=ON"
-    "-DUSE_SYSTEM_CAPSTONE=ON"
-    "-DUSE_SYSTEM_CURL=ON"
-    "-DUSE_SYSTEM_FMT=ON"
-    "-DUSE_SYSTEM_LLVM=ON"
-    "-DUSE_SYSTEM_NLOHMANN_JSON=ON"
-    "-DUSE_SYSTEM_YARA=ON"
+    (lib.cmakeBool "IMHEX_OFFLINE_BUILD" true)
+    (lib.cmakeBool "IMHEX_COMPRESS_DEBUG_INFO" false) # avoids error: cannot compress debug sections (zstd not enabled)
+    (lib.cmakeBool "IMHEX_GENERATE_PACKAGE" stdenv.isDarwin or stdenv.isWindows)
+    (lib.cmakeBool "USE_SYSTEM_CAPSTONE" true)
+    (lib.cmakeBool "USE_SYSTEM_CURL" true)
+    (lib.cmakeBool "USE_SYSTEM_FMT" true)
+    (lib.cmakeBool "USE_SYSTEM_LLVM" true)
+    (lib.cmakeBool "USE_SYSTEM_NLOHMANN_JSON" true)
+    (lib.cmakeBool "USE_SYSTEM_YARA" true)
   ];
 
-  # rsync is used here so we can not copy the _schema.json files
   postInstall = ''
     mkdir -p $out/share/imhex
-    rsync -av --exclude="*_schema.json" ${patterns_src}/{constants,encodings,includes,magic,patterns} $out/share/imhex
+    cp -aR ${patterns_src}/{constants,encodings,includes,magic,patterns} $out/share/imhex
+    find $out/share/imhex -name '*_schema.json' -exec rm {} \;
```

* After building this locally `share/imhex/constants/_schema.json` still exists, I'm uncertain why the find command does not properly remove it 🤔 

* A ~~solution~~workaround for the current darwin build issue is to manually fix the reference to `libimhex` in the app using `install_name_tool`. I'm not particularly fond of doing it this way and maybe someone from @NixOS/darwin-maintainers has a better approach or idea.
```suggestion
    find $out/share/imhex -name '*_schema.json' -exec rm {} \;
  '' + lib.optionalString stdenv.isDarwin ''
    mkdir -p $out/Applications
    mv $out/imhex.app $out/Applications
    install_name_tool \
      -change "$out/lib/libimhex.${finalAttrs.version}${stdenv.hostPlatform.extensions.sharedLibrary}" \
      "@executable_path/../Frameworks/libimhex.${finalAttrs.version}${stdenv.hostPlatform.extensions.
⣀ sharedLibrary}" \
      "$out/Applications/imhex.app/Contents/MacOS/imhex"
```

---

What was the motivation for the removal of rsync? Was it necessary?